src/types/none.rs

```diff
@@ -10,8 +11,11 @@ pyobject_native_type_extract!(PyNone);
 impl PyNone {
     /// Returns the `None` object.
     #[inline]
-    pub fn get(py: Python<'_>) -> &PyNone {
-        unsafe { py.from_borrowed_ptr(ffi::Py_None()) }
+    pub fn get<'py>(py: Python<'py>) -> Borrowed<'py, 'py, PyNone> {
```

Comment:
For backwards-compatibility reasons, we should call this function `get_bound` and deprecate the existing `get` function, like we just did for `PyBool` in https://github.com/PyO3/pyo3/pull/3790/files#diff-5e3c7d536a2349a0e6da69269786babbb9e661764aca7ec9dc304a027ce9fd33